### ISSUE TITLE
Tests using resume_unwind fail with -Cpanic=abort and -Zpanic_abort_tests

### ISSUE THREAD

**Issue by davidhewitt** [rust-lang/rust#74301]
I tried this test code with `RUSTFLAGS="-Cpanic=abort -Zpanic_abort_tests"`:

```rust
    #[test]
    #[should_panic(expected = "new panic")]
    fn resume_unwind_panics() {
        std::panic::resume_unwind(Box::new("new panic"));
    }
```

I expected to see this happen: The test result should be "ok".

Instead, this happened: The test fails:

```
failures:

---- err::tests::resume_unwind_panics stdout ----
---- err::tests::resume_unwind_panics stderr ----
Unexpected error: child process exited with signal 6
```

### Meta

`rustc --version --verbose`:
```
rustc 1.46.0-nightly (daecab3a7 2020-07-10)
binary: rustc
commit-hash: daecab3a784f28082df90cebb204998051f3557d
commit-date: 2020-07-10
host: x86_64-unknown-linux-gnu
release: 1.46.0-nightly
LLVM version: 10.0
```

</p>
</details>

**Comment by davidhewitt** [comment]
I'd be happy to write a patch to resolve this, but I'd need some guidance first as to what a solution might entail. (cc @tmandry ?)

**Comment by tmandry** [comment]
I don't think this is a bug, with the testing machinery at least. The same thing happens when this code is run outside a test in a binary with `-Cpanic=abort`: the process aborts (signal 6) with no information printed.

`resume_unwind` doesn't necessarily translate to `panic` when `panic=abort`, and there's no way to unwind. That said, a friendlier error message would be nice.

**Comment by davidhewitt** [comment]
> I don't think this is a bug, with the testing machinery at least.

That's a reasonable argument. That said, a test suite running with `panic=abort` can fail despite running fine with `panic=unwind`, so it would be desirable to have a way to resolve it.

Perhaps a `#[cfg(not(panic="abort"))]` attribute would be a reasonable way to just remove problematic tests like this?

**Comment by tmandry** [comment]
`panic=abort` changes the semantics of your program, so that's expected in some cases.

I think that'd be reasonable, but as far as I know there isn't a conditional compilation flag for panic behavior (I believe there's an issue for this floating around somewhere). In Fuchsia we manually added a `--cfg` flag for it to our build. I'd support an effort to add a standard one.

**Comment by Enselic** [comment]
Triage: Looks like this was resolved with https://github.com/rust-lang/rust/pull/74754? I'll go ahead and close. Please re-open if I am mistaken.
